### PR TITLE
Merge fluent2-colors -> main

### PR DESCRIPTION
**Pull request by laminesm**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

This PR merges `fluent2-colors` into `main`. 
Overrides, theme changes and overall controls' behaviors were thoroughly tested to make sure nothing regressed. The only issues found (present on both `fluent2-colors` and the new merge branch) were with the `NavigationBar` and `PillButtonBar` not changing to the green theme properly. These issues are fixed in this PR.

Binary change:

| File | Before | After | Delta |
|------|--------|-------|-------|
| libFluentUI.a | 29,253,336 bytes | 29,111,704 bytes | -141,632 bytes |

### Verification

**Fixed Issues**
|Components |fluent2-colors branch | main before                                       | main after                                      |
|----|----|----------------------------------------------|--------------------------------------------|
|NavigationBar| f2colors_nav_bar | before_nav_bar | navbar |
|PillButtonBar| f2colors_pill_button_bar | before_pill_button_bar | Screenshot 2023-02-02 at 1 55 43 PM |

**Demo app controls**
|Components |fluent2-colors branch | main before                                       | main after                                      |
|----|----|----------------------------------------------|--------------------------------------------|
|ActivityIndicator| f2colors_activity_indicator | before_activity_indicator | after_activity_indicator |
|Avatar| f2colors_avatar | before_avatar | after_avatar |
|AvatarGroup| f2colors_avatar_group | before_avatar_group | after_avatar_group |
|CardNudge| f2colors_card_nudge | before_card_nudge | after_card_nudge |
|Global Color Tokens| f2colors_color_tokens | before_color_tokens | after_color_tokens |
|CommandBar| f2colors_command_bar | before_command_bar | after_command_bar |
|HUD| f2colors_hud | before_hud | after_hud |
|NotificationView| f2colors_notification | before_notification | after_notification |
|IndeterminateProgressBar| f2colors_progress_bar | before_progress_bar | after_progress_bar |
|Other cells| f2colors_other_cells | before_other_cells | after_other_cells |
|PersonaButtonCarousel| f2colors_persona_carousel | before_persona_carousel | after_persona_carousel |
|SegmentedControl| f2colors_segmented_control | before_segmented_control | after_segmented_control |
|ShimmerView| f2colors_shimmer | before_shimmer | after_shimmer |
|Tooltip| f2colors_tooltip | before_tooltip | after_tooltip |
|TableViewCell| f2colors_tvc | before_tvc | after_tvc |
|TypographyTokens| f2colors_typography | before_typography | after_typography |
|Alias Color Tokens| f2colors_alias_color_tokens | N/A | after_alias_color_tokens |
|Shadow Tokens| f2colors_shadow_tokens | N/A | after_shadow_tokens |
|BadgeField| f2colors_badgefield | before_badgefield | after_badgefield |
|BadgeView| f2colors_badgeview | before_badgeview | after_badgeview |
|Button| f2colors_button | before_button | after_button |
|BottomCommandingController| f2colors_bcc | before_bcc | after_bcc |
|BottomSheet| f2colors_bottom_sheet | before_bottom_sheet | after_bottom_sheet |
|Shared color variants| N/A | before_color | N/A |
|Card| f2colors_card | before_card | after_card |
|Drawer| f2colors_drawer | before_drawer | after_drawer |
|DateTimePicker| f2colors_date_time_picker | before_date_time_picker | after_date_time_picker |
|Calendar| f2colors_calendar | before_calendar | after_calendar |
|Label| f2colors_label | before_label | after_label |
|Navigation Bar| f2colors_navbar | before_navbar | after_navbar |
|PeoplePicker| f2colors_people_picker | before_people_picker | after_people_picker |
|PersonaListView| f2colors_persona_list_view | before_persona_list_view | after_persona_list_view |
|PillButton| f2colors_pill_button | before_pill_button | after_pill_button |
|PillButtonBar| f2colors_pill_button_bar | before_pill_button_bar | after_pill_button_bar |
|PopupMenu| f2colors_popup | before_popup | after_popup |
|SearchBar| f2colors_searchbar | before_searchbar | after_searchbar |
|SideTabBar| f2colors_side_tab_bar | before_side_tab_bar | after_side_tab_bar |
|TabBar| f2colors_tab_bar | before_tab_bar | after_tab_bar |
|TableViewCellFileAccessoryView| f2colors_tvc_file | before_tvc_file | after_tvc_file |
|HeaderFooter| f2colors_header_footer | before_header_footer | after_header_footer |
|TableViewCellShimmer| f2colors_tvc_shimmer | before_tvc_shimmer | after_tvc_shimmer |
|Objc demo| f2colors_objc_demo | before_objc_demo | after_objc_demo |

### Pull request checklist

This PR has considered:
- [ ] Light and Dark appearances
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)
 ###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://api.github.com/repos/microsoft/fluentui-apple/pulls/1539)

### CODE REVIEW

**Comment by harrieshin** [comment]
what's up with the screenshot of pillbuttonbar and navigation bar? main before is green but main after is not green?

File: ios/FluentUI.Demo/FluentUI.Demo.xcodeproj/xcshareddata/xcschemes/Demo.Release.xcscheme [hunk 1]
```diff
@@ -90,6 +90,17 @@
                ReferencedContainer = "container:FluentUI.Demo.xcodeproj">
             </BuildableReference>
          </TestableReference>
+         <TestableReference
```
Comment (harrieshin) [inline ios/FluentUI.Demo/FluentUI.Demo.xcodeproj/xcshareddata/xcschemes/Demo.Release.xcscheme:93]: are you out of sync? didn't @joannaqu recently remove?

File: ios/FluentUI.Demo/FluentUI.Demo/ColoredPillBackgroundView.swift [hunk 2]
```diff
@@ -27,38 +27,21 @@ class ColoredPillBackgroundView: UIView {
...
+            backgroundColor = UIColor(dynamicColor: fluentTheme.aliasTokens.colors[.background2])
         case .brand:
-            if let fluentTheme = window?.fluentTheme {
-                lightColor = UIColor(dynamicColor: fluentTheme.aliasTokens.backgroundColors[.brandRest])
-                darkColor = UIColor(dynamicColor: fluentTheme.aliasTokens.backgroundColors[.neutral4])
-            } else {
-                lightColor = Colors.communicationBlue
-                darkColor = Colors.navigationBarBackground
-            }
+            backgroundColor = UIColor(dynamicColor: DynamicColor(light: GlobalTokens.brandColors(.comm80),
```
Comment (harrieshin) [inline ios/FluentUI.Demo/FluentUI.Demo/ColoredPillBackgroundView.swift:42]: this is wrong. we should be listening to the branding colors right?

**Comment by laminesm** [comment]
> what's up with the screenshot of pillbuttonbar and navigation bar? main before is green but main after is not green?

The PillButtonBar background is using a global token instead of a brand color and the navigation bar needs to update colors in didMoveToWindow.

File: ios/FluentUI/Table View/TableViewCell.swift [hunk 3]
```diff
@@ -888,7 +883,7 @@ open class TableViewCell: UITableViewCell, TokenizedControlInternal {
         let unreadDotSize = TableViewCellTokenSet.unreadDotDimensions
         unreadDotLayer.bounds.size = CGSize(width: unreadDotSize, height: unreadDotSize)
         unreadDotLayer.cornerRadius = unreadDotSize / 2
-        unreadDotLayer.backgroundColor = UIColor(dynamicColor: tokenSet[.mainBrandColor].dynamicColor).cgColor
+        unreadDotLayer.backgroundColor = UIColor(dynamicColor: isSelected ? tokenSet[.brandTextColor].dynamicColor : tokenSet[.selectionIndicatorOffColor].dynamicColor).cgColor
```
Comment (harrieshin) [inline ios/FluentUI/Table View/TableViewCell.swift:886]: why would unreaddotlayercolor change based on selection state?

File: ios/FluentUI/Table View/TableViewCell.swift [hunk 4]
```diff
@@ -470,6 +463,7 @@ open class TableViewCell: UITableViewCell, TokenizedControlInternal {
     ///   - titleFont: The title font; If not set, it will default to the font definition in tokens
     ///   - subtitleFont: The subtitle font; If not set, it will default to the font definition in tokens
     ///   - footerFont: The footer font; If not set, it will default to the font definition in tokens
+    ///   - hasunreadDot: Boolean determining whether to show or hide the `unreadDotLayer`, on the leading edge of the `customView`.
```
Comment (harrieshin) [inline ios/FluentUI/Table View/TableViewCell.swift:466]: must be a bad merge. @sophialee0416  got rid of this a while a go

File: ios/FluentUI/Date Time Pickers/Date Picker/DatePickerController.swift [hunk 5]
```diff
@@ -488,7 +508,7 @@ extension DatePickerController: CalendarViewLayoutDelegate {
 extension DatePickerController: CalendarViewStyleDataSource {
     func calendarViewDataSource(_ dataSource: CalendarViewDataSource, textStyleForDayWithStart dayStartDate: Date, end: Date, dayStartComponents: DateComponents, todayComponents: DateComponents) -> CalendarViewDayCellTextStyle {
 
-        if dayStartComponents.dateIsTodayOrLater(todayDateComponents: todayComponents) {
+        if dayStartComponents.dateIsInCurrentMonth(todayDateComponents: todayComponents) {
```
[the remark this answers is not in the document]
Comment (laminesm) [reply ios/FluentUI/Date Time Pickers/Date Picker/DatePickerController.swift:511]: It's by design. Old fluent 1 DateTimePicker had different colors for past dates and future dates. On the other hand, fluent 2 DateTimePicker uses different colors for current months and other months (past and future).

File: ios/FluentUI.Demo/FluentUI.Demo.xcodeproj/xcshareddata/xcschemes/Demo.Release.xcscheme (hunk 1 above)
Comment (laminesm) [reply ios/FluentUI.Demo/FluentUI.Demo.xcodeproj/xcshareddata/xcschemes/Demo.Release.xcscheme:93]: #1537 ? If so, it was merged this morning and isn't present on this branch. I can rebase this branch with it.

File: ios/FluentUI.Demo/FluentUI.Demo/ColoredPillBackgroundView.swift (hunk 2 above)
Comment (laminesm) [reply ios/FluentUI.Demo/FluentUI.Demo/ColoredPillBackgroundView.swift:42]: Yes, this is the cause of the PillButtonBar background theming issue. I was thinking about fixing this issue together with the navbar issue in separate PRs. Should I just do it as part of this merge?

Comment (harrieshin) [reply ios/FluentUI.Demo/FluentUI.Demo/ColoredPillBackgroundView.swift:42]: this is a clear regression we should fix

File: ios/FluentUI/Table View/TableViewCell.swift (hunk 3 above)
Comment (laminesm) [reply ios/FluentUI/Table View/TableViewCell.swift]: Seems like it came from a previous merge issue. It's fixed to use the right token.

File: ios/FluentUI/Table View/TableViewCell.swift [hunk 6]
```diff
@@ -883,7 +876,7 @@ open class TableViewCell: UITableViewCell, TokenizedControlInternal {
         let unreadDotSize = TableViewCellTokenSet.unreadDotDimensions
         unreadDotLayer.bounds.size = CGSize(width: unreadDotSize, height: unreadDotSize)
         unreadDotLayer.cornerRadius = unreadDotSize / 2
-        unreadDotLayer.backgroundColor = UIColor(dynamicColor: isSelected ? tokenSet[.brandTextColor].dynamicColor : tokenSet[.selectionIndicatorOffColor].dynamicColor).cgColor
+        unreadDotLayer.backgroundColor = UIColor(dynamicColor: tokenSet[.booleanCellBrandColor].dynamicColor).cgColor
```
Comment (edjamesmsft) [inline ios/FluentUI/Table View/TableViewCell.swift:879]: Is there a reason you chose booleanCellBrandColor over brandTextColor?

**Review by edjamesmsft — APPROVED** [review]

File: ios/FluentUI/Table View/TableViewCell.swift (hunk 6 above)
Comment (laminesm) [reply ios/FluentUI/Table View/TableViewCell.swift:879]: Yep, brandTextColor uses BFG1 but we need BBG1 from booleanCellBrandColor for the unread dot.

File: ios/FluentUI.Demo/FluentUI.Demo.xcodeproj/xcshareddata/xcschemes/Demo.Release.xcscheme (hunk 1 above)
Comment (joannaquu) [reply ios/FluentUI.Demo/FluentUI.Demo.xcodeproj/xcshareddata/xcschemes/Demo.Release.xcscheme:93]: Yep, Mike merged it in this morning. Rebasing will also remove the pr check that's not running right now.

File: ios/FluentUI.Demo/FluentUI.Demo/Demos/TableViewCellDemoController.swift [hunk 7]
```diff
@@ -11,14 +11,6 @@ import UIKit
 class TableViewCellDemoController: DemoTableViewController {
     let sections: [TableViewSampleData.Section] = TableViewCellSampleData.sections
 
-    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?) {
```
Comment (harrieshin) [inline ios/FluentUI.Demo/FluentUI.Demo/Demos/TableViewCellDemoController.swift:14]: why are these deleted?

Comment (laminesm) [reply ios/FluentUI.Demo/FluentUI.Demo/Demos/TableViewCellDemoController.swift:14]: They were removed so that the tableview defaults to its insetGroup style set in DemoTableViewController. See #1535

**Review by harrieshin — APPROVED** [review]
